### DISCUSSION TITLE
Can shift-down automatically add a newline at EOF?

### DISCUSSION THREAD

**Discussion by madeinouweland** [zed-industries/zed#6483] [Feature Requests]
Consider this code of exactly 2 lines:
```
def test():
    pass
```
Selecting this code with SHIFT-DOWN results in a selection that ends after the last character.
To duplicate the function (with the intend to rename the second test to test2), I'd like to press CMD C, CMD V,  CMD V.

But this results in:

```
def test():
    passdef test():
        pass
```

Is it possible to allow SHIFT-SELECTING to pass below the end of file by creating a newline at the end or another smart trick?

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
